File: ts-sdk/src/sdk/models/operations/getshieldinformationbarriersegmentrestrictionsid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class GetShieldInformationBarrierSegmentRestrictionsIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=shield_information_barrier_segment_restriction_id" })
  shieldInformationBarrierSegmentRestrictionId: string;
}


export class GetShieldInformationBarrierSegmentRestrictionsIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: GetShieldInformationBarrierSegmentRestrictionsIdPathParams;
}


export class GetShieldInformationBarrierSegmentRestrictionsIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  shieldInformationBarrierSegmentRestriction?: shared.ShieldInformationBarrierSegmentRestriction;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: sdks/go/pkg/models/operations/getshieldinformationbarriersegmentrestrictionsid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetShieldInformationBarrierSegmentRestrictionsIDPathParams struct {
	ShieldInformationBarrierSegmentRestrictionID string `pathParam:"style=simple,explode=false,name=shield_information_barrier_segment_restriction_id"`
}

type GetShieldInformationBarrierSegmentRestrictionsIDRequest struct {
	Retries    *utils.RetryConfig
	PathParams GetShieldInformationBarrierSegmentRestrictionsIDPathParams
}

type GetShieldInformationBarrierSegmentRestrictionsIDResponse struct {
	ClientError                                *shared.ClientError
	ContentType                                string
	ShieldInformationBarrierSegmentRestriction *shared.ShieldInformationBarrierSegmentRestriction
	StatusCode                                 int64
}
